google/cloud/spanner/internal/date.cc

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/spanner/internal/date.h"
+#include <ctime>
+#include <iomanip>
+#include <sstream>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+
+// All the civil-time code assumes the proleptic Gregorian calendar.
+
+namespace {
+
+// RFC3339 "full-date".
+char const kDateFormat[] = "%Y-%m-%d";
+
+}  // namespace
+
+std::string DateToString(Date d) {
+  std::tm tm;
```

Comment:
Does it matter that `tm` may have uninitialized fields when we call `std::put_time`? We are setting tm_year, tm_mon, and tm_mday, and I *assume* no other fields are accessed, but if they are, they'd be uninitialized.

We could avoid the question with `std::tm tm{};`. Thoughts?

Reply:
As discussed offline, as long as we set the fields that `std::put_time()` is defined to read for the given format specifiers, we're in good shape.

The same is true for `std::get_time()` ... if it succeeds it is OK to read the fields corresponding to the format specifiers ... but clang-tidy doesn't understand that, so unfortunately we have to initialize them first.